File: LabsTechTestTask/ClientApp/src/app/calculator/shared/calculator.service.ts
import { Injectable } from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {Observable} from 'rxjs';
import {environment} from '../../../environments/environment';
import {CalculationRequest} from './calculator-request.model';
import {CalculatorResponse} from './calculator-response.model';

@Injectable({
  providedIn: 'root'
})
export class CalculatorService {

  constructor(public http: HttpClient) {
  }

  calculate(request: CalculationRequest): Observable<CalculatorResponse> {
    return this.http.get<CalculatorResponse>(`${environment.apiUrl}/api/calculator?operator1=${request.operator1}&operand=${request.operand}&operator2=${request.operator2}`);
  }

}

File: LabsTechTestTask.BAL/Services/Implementations/CalculatorService.cs
using LabsTechTestTask.BAL.Implementations.Calculator.Common;
using LabsTechTestTask.BAL.Models.Calculator;
using LabsTechTestTask.BAL.Services;
using LabsTechTestTask.Models;
using System;
using System.Collections.Generic;
using System.Text;

namespace LabsTechTestTask.BAL.Implementations
{
    public class CalculatorService : ICalculatorService
    {

        public CalculatorResponse Calculate(CalculationRequest request)
        {
            var operation = OperationFactory.Create(request.Operand);

            var result = operation.Calculate(request.Operator1, request.Operator2);

            return new CalculatorResponse { Result = result };
        }

    }
}
